storage/remote/queue_manager.go

```diff
@@ -305,8 +305,12 @@ type QueueManager struct {
 	seriesSegmentIndexes map[uint64]int
 	droppedSeries        map[uint64]struct{}
 
-	shards      *shards
-	numShards   int
+	shards    *shards
+	numShards int
```

Comment:
This all needs locking

Reply:
Yeah, I was just roughly drafting out on the structs to get a feel. Need to go back and make sure appropriate mutexes are added.